### PR TITLE
Updated the Windows ScopedAutoTempDirectory creation logic to use a Uuid

### PR DESCRIPTION
**Pull request by lumberyard-employee-dm**
Previously it was using `GetTickCount()` for the creation of the
directory which can collide if  there are multiple processes
creating a temporary directory via the `ScopedAutoTempDirectory`
constructor.

Signed-off-by: lumberyard-employee-dm <56135373+lumberyard-employee-dm@users.noreply.github.com>

### CODE REVIEW

**Review by amznestebanpapp — APPROVED** [review]

**Review by onecent1101 — APPROVED** [review]

File: Code/Framework/AzTest/AzTest/Platform/Windows/ScopedAutoTempDirectory_Windows.cpp [hunk 1]
```diff
@@ -8,57 +8,41 @@
...
+        AZStd::to_string(tempDirectoryRoot.Native(), tempDirW);
 
-            int maxAttempts = 2000; // Prevent an infinite loop by setting an arbitrary maximum attempts at finding an available temp folder name
-            while (maxAttempts > 0)
+        constexpr int MaxAttempts = 255;
+        for (int i = 0; i < MaxAttempts; ++i)
+        {
+            AZ::IO::FixedMaxPath testPath = tempDirectoryRoot /
+                AZ::IO::FixedMaxPathString::format("UnitTest-%s",
+                    AZ::Uuid::CreateRandom().ToString<UuidString>().c_str());
```
Comment (spham-amzn) [inline Code/Framework/AzTest/AzTest/Platform/Windows/ScopedAutoTempDirectory_Windows.cpp:35]: Almost feels like a mutex is needed somewhere during the random generation and creation to be extra safe, but probably not

**Review by spham-amzn — APPROVED** [review]

File: Code/Framework/AzTest/AzTest/Platform/Windows/ScopedAutoTempDirectory_Windows.cpp (hunk 1 above)
Comment (lumberyard-employee-dm) [reply Code/Framework/AzTest/AzTest/Platform/Windows/ScopedAutoTempDirectory_Windows.cpp:35]: The [Uuid::CreateRandom](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/Uuid.cpp#L249-L252) function uses the [Sfmt::Rand32](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/Sfmt.cpp#L516-L521) function which already provides mutex protection against the generation of the random number.

Comment (spham-amzn) [reply Code/Framework/AzTest/AzTest/Platform/Windows/ScopedAutoTempDirectory_Windows.cpp:35]: Yeah, i was thinking of the whole create random -> create folder block (sorry didnt highlight the whole section). But random generation is done through a sequence anyways, so probably would never be an issue, so it wouldnt really matter anyways

Comment (lumberyard-employee-dm) [reply Code/Framework/AzTest/AzTest/Platform/Windows/ScopedAutoTempDirectory_Windows.cpp:35]: Well there are no shared variables that could run on multiple threads.

The temporary directory creation is in a constructor, which can only occur on a single thread.

**Pull request merged**